### PR TITLE
Enable new vis_cpu coordinate system treatment

### PR DESCRIPTION
**Pull request by philbull**
This PR enables the new coordinate system handling features for VisCPU associated with this PR:
https://github.com/HERA-Team/vis_cpu/pull/11

This includes a method that allows more accurate handling of equatorial coordinates, and has more explicit documentation on what coordinates are assumed. This also enables basic polarization support.

Target: v1.0.0.

### CODE REVIEW

File: setup.cfg
Comment (steven-murray) [inline setup.cfg:46]: I would leave this with the >=, since we really do need that version or greater.

File: setup.cfg
Comment (steven-murray) [inline setup.cfg:73]: I couldn't see what IPython was being used for?

File: setup.cfg
Comment (steven-murray) [inline setup.cfg:78]: We're trying to release hera_sim as a PyPI package, and as such it can't have git-url requirements like this (I've made issues on all the other repos above this to this effect as well). We should just release a new vis_cpu version so that this branch works with the PyPI version of vis_cpu.

**Review by steven-murray — COMMENTED** [review]
Thanks @philbull -- just a few fixes/clarifications to be made. I can release a new vis_cpu if that helps as well.

File: hera_sim/tests/test_vis.py
Comment (philbull) [reply hera_sim/tests/test_vis.py:245]: This is the fix I guess -- I switched vis_cpu to applying a factor of 1/2 to the input source fluxes (assumed to be specified as Stokes I), as explained here: https://github.com/HERA-Team/vis_cpu/blob/main/src/vis_cpu/vis_cpu.py#L92
The rest of vis_cpu naturally deals with linear polarisation channels (rather than Stokes), so the output in this case is for the 'xx' polarisation and so needs to be multiplied by a factor of 2 again to get Stokes I.

This change makes the output consistent with pyuvsim, which you can see from this unit test in the vis_cpu package: https://github.com/HERA-Team/vis_cpu/blob/main/tests/test_compare_pyuvsim.py#L143

But in _this_ unit test (in hera_sim/tests/test_vis.py), the extra factor of 2 is needed, as I assume the other simulators are outputting Stokes I.

So, I'm not sure what the best course of action is here. I think the new vis_cpu behaviour makes sense, as it is now a polarised visibility simulator, and it matches pyuvsim.

File: setup.cfg
Comment (philbull) [reply setup.cfg:46]: Thanks, I've actually bumped this to >=0.2.0, since this PR uses recent changes in vis_cpu

File: setup.cfg
Comment (philbull) [reply setup.cfg:78]: Thanks, this was added in this PR just to get the tests to work properly. Now the vis_cpu version has been bumped I've removed it.

File: hera_sim/tests/test_vis.py
Comment (steven-murray) [reply hera_sim/tests/test_vis.py:245]: Hmm, OK if vis_cpu is consistent with pyuvsim, it probably means we instead need to update the HealVis wrapper to return results consistent with pyuvsim. This has been a persistent problem, and has not been helped by my own very vague understanding of polarization etc.

**Review by steven-murray — APPROVED** [review]
Happy for this to go in -- but if you feel like taking a look at the HealVis wrapper, it might be worth fixing that up and then changing these tests back so that no *2 is required. Otherwise, we can create an issue for it and resolve it independently.

File: hera_sim/tests/test_vis.py
Comment (philbull) [reply hera_sim/tests/test_vis.py:245]: Thanks, I can look into this if needed, but perhaps it's more sensible for the HealVis maintainer to do it (not sure who at the moment?)

File: setup.cfg
Comment (philbull) [reply setup.cfg]: This was mistakenly included. There was an issue with vis_cpu importing IPython, but not actually including it properly in its install dependencies. I have now made it optional in this PR: https://github.com/HERA-Team/vis_cpu/pull/13

**Comment by philbull** [comment]
Thanks Steven. I think it's best to merge this in now if you're happy with it, and then look at HealVis separately, as in my experience anything to do with polarisation is a can of worms...

**Pull request merged**
